clients/drcachesim/tracer/tracer.cpp

```diff
@@ -1021,14 +1033,17 @@ is_last_instr(void *drcontext, instr_t *instr)
  */
 static dr_emit_flags_t
 event_app_instruction(void *drcontext, void *tag, instrlist_t *bb, instr_t *instr,
-                      instr_t *where, bool for_trace, bool translating,
+                      instr_t *where, bool for_trace, bool translating, uintptr_t mode,
                       void *orig_analysis_data, void *user_data)
 {
     int i, adjust = 0;
     user_data_t *ud = (user_data_t *)user_data;
     reg_id_t reg_ptr;
     drvector_t rvec;
     dr_emit_flags_t flags = DR_EMIT_DEFAULT;
+    bool is_l0_filter = (op_L0I_filter.get_value() || op_L0D_filter.get_value());
+    if (need_l0_filter_mode)
```

Comment:
I'm not sure I understand this variable need_l0_filter_mode which is never set in this PR.   I assume since it overrides the op_L0*_filter options the plan is to set it for warmup and then unset it, but wouldn't the `mode` parameter now passed here serve that same purpose?  I.e., it seems we do not need this need_l0_filter_mode?  Seeing how this is used would help clarify.  I would suggest adding the mechanism to set it in this PR, and adding a test in the PR as well.

Reply:
I removed the need_l0_filter_mode variable and added a cmd line option. Hope it looks better now. Right now I have not implemented changes to switch from the warmup mode to trace mode, so the feature is quite incomplete. Should I add tests just for the warmup mode creation or wait until the full feature is done?

Reply:
Yes, it is better IMHO.  Since the switch from warmup to full should be relatively little code (unless I'm missing something) I would suggest just adding that to this PR too and adding a test?

Reply:
See PR #5742 where a marker delineating the filter-full boundary is being added (there by offline editing): it makes sense to use that same marker here.